Lib/test/test_importlib/test_util.py

```diff
@@ -758,6 +760,94 @@ def test_source_from_cache_path_like_arg(self):
         expect = os.path.join('foo', 'bar', 'baz', 'qux.py')
         self.assertEqual(self.util.source_from_cache(path), expect)
 
+    @contextlib.contextmanager
+    def bytecode_path(self, path):
+        # Patch directly in globals of the functions under test, because we have
+        # no direct access to the right _bootstrap_external module to patch
+
+        # cache_from_source and source_from_cache come from the same module,
+        # thus they share the same globals dict, so we only have to patch it
+        # once
+        _globals = self.util.cache_from_source.__globals__
+
+        _orig_path = _globals['bytecode_path']
+        _globals['bytecode_path'] = path
+        try:
+            yield
+        finally:
+            _globals['bytecode_path'] = _orig_path
+
+    @unittest.skipUnless(sys.implementation.cache_tag is not None,
+                         'requires sys.implementation.cache_tag not be None')
+    def test_cache_from_source_respects_PYTHONBYTECODEPATH(self):
+        # If PYTHONBYTECODEPATH env var is set, cache_from_source will return
+        # a bytecode path inside that directory (in a subdirectory mirroring the
+        # .py file's path) rather than in a __pycache__ dir next to the py file.
+        bytecode_paths = [
+            os.path.join(os.path.sep, 'tmp', 'bytecode'),
```

Comment:
These have a definite posix vibe to them.  Could you also mix in some Windows-specific paths too.  Note that the semantics of such will be different depending on platform (see my note above on cache_from_source), so the tests will need to differ depending on platform.

Reply:
These functions (`source_from_cache` and `cache_from_source`) and their tests only do path manipulation, not path access. I don't intend to add any tests that do filesystem access, since that would be testing existing behavior not modified by this PR. So I don't think I will need to gate on access.

Ultimately the only thing that matters for this PR is that we correctly detect UNC paths as "absolute" and don't try to prepend CWD to them. It's clear from inspection that we do that (since they begin with a path separator), but I agree it would be good to have it tested; will add tests.